hasura/graphql-engine
# refactor AddTable validation, fix #1712

### Description
This PR is a refactor of submitValidation and it's helper, columnValidation, which had become a bit of spaghetti, and which had bug #1712  
Inspecting code found several others.
I have rewritten the code to be cleaner and more uniform in structure, in the process fixing these bugs.

### Affected components 
- Console


### Related Issues
#1712

### Solution and Design
Original code had a nest of logic to do what really is simply a series of checks.
Now simply does the checks in order.
Additionally, I have refactored the error checking to separate the error test and the
dispatch of an error so the logic is clearer.

### Steps to test and verify
See steps in #1712 
1. make a table named author with these columns
name=author  type= Integer  (auto increment) check unique
name=name  type=<leave default, none selected>

2. submit - 
observe an error message instead of the server side message

### Limitations, known bugs & workarounds
There is no code support for validating postgres type literals on the client side.
Rather than extend this fix to fixing this issue, I pass everything (which is current behavior),
and reported https://github.com/hasura/graphql-engine-internal/issues/199
